File: P134.py
n = int(input())
a = [0] + list(map(int,input().split()))
chan = [0] * (n+1)
le = [0] * (n+1)
ans = 0
for i in range(1,n+1):
    chan[i] = chan[i-1]
    le[i] = le[i-1]
    if a[i] % 2 == 0: chan[i] += 1
else: le[i] += 1
for i in range(1,n+1):
    for j in range(i+1,n+1):
        if chan[j] - chan[i-1] == le[j]-le[i-1]:
            ans = max(ans, j-i+1)
print(ans)

File: P134.cpp
#include <bits/stdc++.h>
#define oo 1000000007
#define pb push_back
#define ff first
#define ss second
#define mp make_pair
#define ll long long
#define ii pair <int, int>
#define fto(i, a, b) for (int i = a; i <= b; ++i)
#define fdto(i, a, b) for (int i = a; i >= b; --i)
#define fastio ios_base::sync_with_stdio(false); cin.tie(NULL);
using namespace std;
int n, a[1005], c, l, ans;

int main(){
    fastio;
    cin >> n;
    fto(i, 1, n) cin >> a[i];
    fto(i, 1, n) {
        l = 0;
        c = 0;
        fto(j, i, n) {
            if (a[j] & 1) ++l;
            else ++c;
            if (l == c) ans = max(ans, j-i+1);
        }
    }
    cout << ans;
}